rfcs/20230210-sparsity.md

```diff
@@ -104,35 +104,33 @@ that was also adopted in MLIR for sparse tensor types. Below we use
 We propose to extend the regular tensor types of StableHLO
 with a new encoding field that allows for providing:
 
-1. An ordered sequence of level specifications, each of which includes:
-   1. a required **level-type**, which defines *how* the level is stored,
-      including:
-      1. a required **level-format**
-      2. a collection of **level-properties**
-   2. a **level-expression**, which defines *what* is stored.
-2. An ordered sequence of dimension specifications, each of which includes:
+1. An ordered sequence of dimension specifications, each of which includes:
    1. the **dimension-size**; however, that’s already given by the tensor’s
       dimension-shape
    2. an optional **dimension-expression**
+2. An ordered sequence of level specifications, each of which includes:
+   1. a **level-expression**, which defines *what* is stored.
+   2. a required **level-type**, which defines *how* the level is stored,
+      including:
+      1. a required **level-format**
+      2. a collection of **level-properties**
 
 Different level-formats may have different collections of level-properties,
 and whether these properties are optional vs required also depends on the
 level-format; hence why we group the level-format and level-properties
 together as a level-type. Whether the level-expression is optional vs
 required may also depend on the level-type. For the three level-formats
 discussed below, both the level-properties and level-expressions are
-optional. (Whereas the level-format for supporting `ITPACK/ELLPACK`
-has no properties and the level-expression is required.)
+optional.
 
-For the three level-types discussed below, each level-expression is an
+For the level-types discussed below, each level-expression is an
 affine expression over dimension-variables. Thus, the level-expressions
 collectively define an affine map from dimension-coordinates to
 level-coordinates. And the dimension-expressions collectively define the
 inverse map— which only needs to be provided for elaborate cases

```

Comment:
One more nit. Should we remove the - here?

Reply:
Yes, we should indeed!